### ISSUE TITLE
Generate a parity list report

### ISSUE THREAD

**Issue by VividLemon** [bootstrap-vue-next/bootstrap-vue-next#1775]
### Clear and concise description of the problem

In order to release v1, a parity list spreadsheet needs to be created for internal use. A list comparing the feature set of bv to bvn, including: props, emits, and functionality. 

This is vital to the release of the library as we need to be able to properly index what is missing and what is simply removed behavior. 

Even things that are declared deprecated in bv should be noted on the parity list, but have a comment describing they are deprecated.

An online spreadsheet (please do not post direct files here, for security) as an evolving spreadsheet.

### Suggested solution

An online spreadsheet should be created here to document every components parity with that of bv. It should contain information about prop differences, (say if bv had type `"string"` and we had type `"sm" | 'md' | 'lg'`. It's not invalid, but it needs to be documented)

### Alternative

This needs to be finished by the time v1 is released

### Validations

- [X] Follow our [Code of Conduct](https://github.com/bootstrap-vue-next/bootstrap-vue-next/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/bootstrap-vue-next/bootstrap-vue-next/blob/main/CONTRIBUTING.md).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

**Comment by VividLemon** [comment]
Note: this is related to https://github.com/bootstrap-vue-next/bootstrap-vue-next/issues/1780 , in addition to a parity list with bvn, we must also review new features that were added to Bootstrap v5, that have not been implemented in bvn.

For example, stripedColumns were an addition to bootstrap v5, but weren't added until like v0.12. These things need to be documented as well.

**Comment by dwgray** [comment]
I've been doing something thinking about this and have a proposed first pass at a parity spreadsheet.   One of the things that I iterated over on this was the granularity.  I landed on having a row for each component, property, event, and slot (as well as a grab-bag type of row for things that don't fall into any of those, like a Bootstrap5 feature that's not yet implemented in this system).  This seemed like a reasonable middle road.  If we want to just have a row per component, it might be better to just have an issue for each component, with a parity-report tag and possibly a separate issue template.  If we want to go crazy on this, we might be able to create rows for event arguments and slot-scoped properties, but that feels like a bit much.

I'm starting with just components. I realize that there are other things in the code and documentation that need to be tracked - I think that we may be able to fold most of them into this spreadsheet as different "kinds," but I'd like to get a handle on components first.  We might also have a separate tab for other things.

In any case, happy to adjust this based on feedback.  My next step once we have consensus on a spreadsheet format would be to auto-populate an initial version with the BSV legacy items, at which point we can start picking away at the individual components.

I did this in Excel, hosted on OneDrive, which is what I'm most comfortable with.  If there is a strong sentiment for some alternative, I'm happy to try that.  The following link should be viewable to anyone.  I'll obviously give contributors write access once we get things nailed down.  There are two tabs, one defining the columns in the report and the other the actual report.

https://1drv.ms/x/s!AiUqzkjNYGL6ieBPpQpcR41wo1laZQ?e=aebW4W

And a couple of screenshots for easier access.

![image](https://github.com/bootstrap-vue-next/bootstrap-vue-next/assets/5083020/3e279630-33ca-4b5a-8d5b-778052f3858e)

A little bit of sample data:
![image](https://github.com/bootstrap-vue-next/bootstrap-vue-next/assets/5083020/5afc575f-c045-4b5f-8d22-0dad93ffefba)

**Comment by dwgray** [comment]
I've taken a stab at creating a spreadsheet for tracking parity/completion of features.  I hope the spreadsheet is self-explanatory,  The first tab is definitions, and the second is the actual tracking.

Design decisions, which I'm happy to take feedback on:

1. I'm just covering components now.  I believe other things will have somewhat different schemas and could be represented by different tabs, but if we get a good handle on components and their documentation that will get us 80% of the way there.
1. Granularity - Possibilities
    1. One row per component - this seems light.  If we want to do this level, it would make more sense to have an issue per component, possibly with a different template
    1. Row **per component, property, event, slot**, with room for extra rows if there are things about a component that need to be called out.  This is where I landed.
    1. In addition to the above, also have rows for event arguments, slot-scoped properties, etc.  This feels like it will make the schema too complex and would be unmanageable
1. I used Excel on OneDrive, since that's what I'm most familiar with.  Happy to move to a different system if there is a strong sentiment/good reason to do so.
	
A read only  version of the spreadsheet is here: https://1drv.ms/x/s!AiUqzkjNYGL6ieBPpQpcR41wo1laZQ?e=DYykFl - I'll share write access to contributors once we get things locked down.

![image](https://github.com/bootstrap-vue-next/bootstrap-vue-next/assets/5083020/ce273c3b-f6f9-48db-8513-636fd512b566)

![image](https://github.com/bootstrap-vue-next/bootstrap-vue-next/assets/5083020/777c55ee-dde8-47e2-a957-b81755724321)

P.S. I posted something like this yesterday.  When I looked at it again it looked like GitHub duplicated it, so I deleted the duplicate.  But it appears that it disappeared.  So, if the apparent duplicate was someone quoting and commenting, I apologize and hope I didn't delete too much information.  Please go ahead and repost, and I promise not to delete it again….

**Comment by VividLemon** [comment]
>  Please go ahead and repost, and I promise not to delete it again….

It's unlikely that you'd be able to delete other persons posts.

The format looks good, this is likely how I would do it. Do keep in mind that very soon https://github.com/bootstrap-vue-next/bootstrap-vue-next/pull/1791 will be merged. So, some component definitions may change

**Comment by dwgray** [comment]
> The format looks good, this is likely how I would do it. Do keep in mind that very soon #1791 will be merged. So, some component definitions may change

Thanks - my next step is going to be to see if I can figure out an automated way to populate the spreadsheet with the BSV Legacy information, that seems like it would give us a good framework.  I think we probably want human eyes when filling in any of the BSVN information so I'm less convinced that automating that makes sense, but I'll think about that while I get the BSV Legacy step in.

**Comment by dwgray** [comment]
I populated the [parity spreadsheet](https://1drv.ms/x/s!AiUqzkjNYGL6ieBPpQpcR41wo1laZQ?e=DYykFl) with the bootstrap vue legacy information (nearly 2000 rows) and over 100 components!  I wrote automation to do this, and I believe I got everything, but could use another set of eyes to check if I missed anything obvious.

I then went through and filled in the rows for BFormCheckbox and BFormCheckboxGroup since those are components that I've fixed bugs in and updated documentation for recently; note that I filled this out with the assumption that #1791 had already been done.

I added a tab to the spreadsheet with instructions, which includes the process I used as a suggested process.  I'm happy to refine that based on suggestions from others with more experience.  Also interested if I missed anything big.

![image](https://github.com/bootstrap-vue-next/bootstrap-vue-next/assets/5083020/9f5f3d85-399d-4b47-839a-ddffb5d202f2)

Things that still need to be decided before using this to evaluate more components:

Do we want to auto-populate the BSVN columns?  I'm leaning towards no.  I feel like the person filling out a component is going to be spending enough time reconciling the various places where items are used/defined/documented that manually adding these cells won't be much of a burden.  But it also won't be too hard to generate this data - the most challenging part might be weaving the two sets of data together.

I think we'll want to consider what to do about things like button-variant and models. These are things where we generally have parity but are handled a bit differently in BSVN.  I would suggest that we write a general discussion of these differences in the migration guide, then when going through this spreadsheet, if that general discussion seems to cover the difference, mark the migration column "Not Needed."  If that makes sense, I'll add an issue/issues for documenting each of these general issues in the migration guide (and then see if I can find time to do a first pass at them).

How thorough do we want to be about documenting types?  There are several types used in these components that don't show up in the types documentation; should we track that here as well?

How do we want to open up access to this?  I think I'd be all right with just publishing an editable link to this spreadsheet would be easiest.  OneDrive manages versioning, so if a bad actor or an over-enthusiastic novice messes it up, I can just go back to a previous version.  Alternatively, I can add specific users to write access.  Let me know.

**Comment by VividLemon** [comment]
> How thorough do we want to be about documenting types? There are several types used in these components that don't show up in the types documentation; should we track that here as well?

Likely https://github.com/bootstrap-vue-next/bootstrap-vue-next/issues/1585 should be reopened. If they're used, they should likely be exported.

> How do we want to open up access to this? I think I'd be all right with just publishing an editable link to this spreadsheet would be easiest. OneDrive manages versioning, so if a bad actor or an over-enthusiastic novice messes it up, I can just go back to a previous version. Alternatively, I can add specific users to write access. Let me know.

This is a good question with no great answer. Github does not have any builtin way to make an excel sheet, which makes sharing around difficult. Likely only myself and @xvaara need access. Although, anyone that I suppose comments here could also be given access

Alternatively, it may be best to have each file be readonly, where anyone that wishes to make a change would download the sheet, make modifications, then reupload to this thread. However, I do wonder if there's security issues involved with this. 

Overall, very good work. I don't have any specific mentions about the structure or anything else. Migration is definitely more difficult when we have three different places of changes (bootstrap, bv, & vue). Thank you

**Comment by dwgray** [comment]
> > How thorough do we want to be about documenting types? There are several types used in these components that don't show up in the types documentation; should we track that here as well?
> 
> Likely #1585 should be reopened. If they're used, they should likely be exported.
> 
Do we also expect that every exported type is documented on the [types page](https://bootstrap-vue-next.github.io/bootstrap-vue-next/docs/types.html)?  I would say yes.  In which case, maybe we can keep it simple and make verifying that part of the docs complete bar?

> > How do we want to open up access to this? I think I'd be all right with just publishing an editable link to this spreadsheet would be easiest. OneDrive manages versioning, so if a bad actor or an over-enthusiastic novice messes it up, I can just go back to a previous version. Alternatively, I can add specific users to write access. Let me know.
> 
> This is a good question with no great answer. Github does not have any builtin way to make an excel sheet, which makes sharing around difficult. Likely only myself and @xvaara need access. Although, anyone that I suppose comments here could also be given access
> 
> Alternatively, it may be best to have each file be readonly, where anyone that wishes to make a change would download the sheet, make modifications, then reupload to this thread. However, I do wonder if there's security issues involved with this.
> 
So it turns out that onedrive will let me create a separate read/write link that I could share privately with you and @xvaara and you could share out with others at your discretion.  I could also delete that link if it leaks for some reason.  Seems like a decent solution if you're both good with that.  Excel Online is getting pretty good, so you probably don't have to download the file, but if only the three of us have access, it's probably not a _giant_ security risk.  Let me know your preferred channel for me to send you the link - it looks like there is a little used Discord server for this project, so I could DM you on Discord (I saw Xvaara's handle show up in the general channel, but it wasn't obvious what yours is).

> Overall, very good work. I don't have any specific mentions about the structure or anything else. Migration is definitely more difficult when we have three different places of changes (bootstrap, bv, & vue). Thank you

Thanks!

**Comment by euoia** [comment]
What is the best way to contribute updates to the parity report? I just had a quick look and the bTable `items` prop shows a type of Array but it can (importantly) be an Array or a Function.

<img width="1580" alt="image" src="https://github.com/bootstrap-vue-next/bootstrap-vue-next/assets/1271216/d395685e-6185-478f-ad30-d300266b523d">

<img width="909" alt="image" src="https://github.com/bootstrap-vue-next/bootstrap-vue-next/assets/1271216/fc9538ff-272e-4715-bb9a-512356644734">

**Comment by dwgray** [comment]
@euoia I reuploaded the parity spreadsheet with the new data which includes the "union" types.  I looked through the BTable section and things appear to line up.  Please let me know if you see anything else I missed.  Thanks!

**Comment by VividLemon** [comment]
All of the last comments are things I agree with. For some reason I've been forgetting about this, and didn't realize that it had gotten so old. I am okay with a general spreadsheet being passed around. It seems like the quickest way to have this issue fixed, as this is very high importance for the prospects of v1. Seems loosening restrictions is a necessity for quickness. The only reservations I have are those security related in excel itself. And you never really know with it being truly open. But it seems like a low risk high reward situation, if any risk

My discord is @templeos

**Comment by dwgray** [comment]
I've put up a [PR](https://github.com/bootstrap-vue-next/bootstrap-vue-next/pull/1834/files) to update `contributing.md` to point folks at the feature parity spreadsheet (and hopefully get more folks to contribute to it).  If this looks good, I'll add a pointer to that in the "how can I help" and "when is this going to be released" threads.

**Comment by dwgray** [comment]
@VividLemon - discord wouldn't let me DM you with the password for the r/w version of the spreadsheet.  I sent you a friend request, my handle is @ohdwg, which seems like you may require to DM?

**Comment by VividLemon** [comment]
Deprecation note https://github.com/bootstrap-vue-next/bootstrap-vue-next/pull/1830#issuecomment-2032773290

**Comment by VividLemon** [comment]
Deprecation note https://github.com/bootstrap-vue-next/bootstrap-vue-next/issues/1798#issuecomment-2032781026

**Comment by VividLemon** [comment]
Deprecation note `BTable` modelValue -> @change event

**Comment by dwgray** [comment]
Based on a [this discord discussion](https://discord.com/channels/335455578921107456/335455578921107456/1222435910768136262) it looks like some people are finding the parity report and using it to assess where the project is. I added some text in a couple of places to try to discourage this until we've gotten farther along. However, people don't always read the docs :-) So I spent some time going through the top-level components and getting their status updated to the best of my ability. Hopefully that will help as well...

**Comment by VividLemon** [comment]
Deprecation note BTable 'selected' event => @update:selectedItems

**Comment by VividLemon** [comment]
I'm not sure any of them are really needed for a reasonable v1, I'd put the more systemic issues that @VividLemon is tracking in the project above any new components.

The only issue with v1 is stability. Not necessarily compatibility.

I will be looking closer at this issue soon

**Comment by VividLemon** [comment]
BFormFile with various changes to what you can set to the input https://github.com/bootstrap-vue-next/bootstrap-vue-next/issues/2052 . Notably setting value to `[]`

**Comment by VividLemon** [comment]
html props are removed https://github.com/bootstrap-vue-next/bootstrap-vue-next/pull/2311
